mmrazor/models/mutables/mutable_channel/groups/channel_group.py

```diff
@@ -0,0 +1,730 @@
+            index (Union[None, Tuple[int, int]]): the channel range for pruning
+            out_related (Bool): represents if the channels are output channels,
+            otherwise input channels
+            expand_ratio (Bool): expand_ratio of the number of channels
+            compared with pruning mask
+        """
+        self.name = name
+        self.module: DynamicChannelMixin = module
+        self.index = index
+        self.start = index[0]
+        self.end = index[1]
+
+        self.node = node
+
+        self.output_related = out_related
+        self.expand_ratio = expand_ratio
+
+    @classmethod
+    def init_using_cfg(cls, model: nn.Module, config: Dict):
+        """init a Channel using a config which can be generated by
+        self.config_template()"""
+        name = config['name']
+        start = config['start']
+        end = config['end']
+        expand_ratio = config['expand_ratio']
+        is_output = config['is_output_related']
+
+        name2module = dict(model.named_modules())
+        name2module.pop('')
+        module = name2module[name] if name in name2module else None
+        return Channel(
+            name,
+            module, (start, end),
+            out_related=is_output,
+            expand_ratio=expand_ratio)
+
+    # config template
+    def config_template(self):
+        """Generate a config template which can be used to initialize a Channel
+        by cls.init_using_cfg(**kwargs)"""
+        return {
+            'name': self.name,
+            'start': self.start,
+            'end': self.end,
+            'expand_ratio': self.expand_ratio,
+            'is_output_related': self.output_related
+        }
+
+    # basic properties
+
+    @property
+    def num_channels(self) -> int:
+        """Int: number of channels in the Channels"""
+        return self.index[1] - self.index[0]
+
+    @property
+    def is_prunable(self) -> bool:
+        """If the channel is prunable."""
+        if isinstance(self.module, nn.Conv2d):
+            # group-wise conv
+            if self.module.groups != 1 and not (self.module.groups ==
+                                                self.module.in_channels ==
+                                                self.module.out_channels):
+                return False
+        return True
+
+    # node operations
+
+    def slice(self, start: int, end: int) -> 'Channel':
+        """Channel: a new Channel who manage a slice of the current Channel."""
+        channel = Channel(
+            name=self.name,
+            module=self.module,
+            index=(self.start + start, self.start + end),
+            node=self.node,
+            out_related=self.output_related,
+            expand_ratio=self.expand_ratio)
+        return channel
+
+    # others
+
+    def __repr__(self) -> str:
+        return f'{self.name}\t{self.index}\t \
+        {"out" if self.output_related else "in"}\t\
+        expand:{self.expand_ratio}'
+
+
+@MODELS.register_module()
```

Comment:
Use `TASK_UTILS` registry

Reply:
I remove the register_module for  ChananelGroup, as we don't need to initialize ChannelGroup using .build().